include/solvers/tracer.h

```diff
@@ -383,6 +423,120 @@ class Tracer : public AuxiliaryPhysics<dim, GlobalVectorType>
   void
   write_tracer_statistics();
 
+
+
+  /**
+   * @brief Get the lethe boundary indicator for a given triangulation boundary while carrying the appropriate checks
+   *
+   * @param[in] triangulation_boundary_id The boundary id of the triangulation
+   *
+   * NOTE: This function is a temporary function. It will be deprecated in a
+   * future PR once boundary conditions have been refactored using maps.
+   */
+  unsigned int
+  get_lethe_boundary_index(const types::boundary_id &triangulation_boundary_id)
+  {
+    // Identify which boundary condition corresponds to the boundary id. If
+    // this boundary condition is not identified, then exit the simulation
+    // instead of assuming an outlet.
+    const auto lethe_boundary_id = std::find(
+      this->simulation_parameters.boundary_conditions_tracer.id.begin(),
+      this->simulation_parameters.boundary_conditions_tracer.id.end(),
+      triangulation_boundary_id);
+
+    AssertThrow(
+      lethe_boundary_id !=
+        this->simulation_parameters.boundary_conditions_tracer.id.end(),
+      ExcMessage("The boundary condition with id " +
+                 std::to_string(triangulation_boundary_id) +
+                 " is not present in the tracer boundary conditions."));
+
+    return (lethe_boundary_id -
+            this->simulation_parameters.boundary_conditions_tracer.id.begin());
+  }
+
+  /**
+   * @brief Helper function to reinit the face velocity with the adequate solution.
+   * This prevents code duplication throughout the tracer. The function looks at
+   * the multiphysics interface to decide if the velocity is a block velocity or
+   * a regular velocity. Furthermore, it also checks if a time-averaged solution
+   * is required. Otherwise the code here would be copied four times.
+   *
+   * @param[in] velocity_cell the iterator to the cell where the velocity is to
+   * be reinitialized.
+   *
+   * @param[in] face_no the number of the face where the velocity is to be
+   * reinitialized.
+   *
+   * @param[in,out] scratch_data the scratch data to be used for the
+   * reinitialization.
+   */
+  inline void
+  reinit_face_velocity_with_adequate_solution(
+    const typename DoFHandler<dim>::active_cell_iterator &velocity_cell,
```

Comment:
I know this is present in other parts of the code in order to use the average velocity or not. Is there an easy way to get rid of sections like this one scattered in the code? I feel it takes a lot of place and could simplify some sections.